tests/rbd/test_rbd_dm_cache.py

```diff
@@ -0,0 +1,175 @@
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.lvm_utils import lvconvert, lvm_create, pvcreate, vgcreate
+from utility.utils import run_fio, run_mkfs
+
+log = Log(__name__)
+
+
+def test_configure_cache(rbd_obj, client_node, pool_name, image_name, cache_type, **kw):
+    """Method to configure cache mechansim as dm-cache
+    and dm-write cache for RBD image based devices.
+
+    Args:
+        rbd_obj: RBD object
+        client_node: cache client node
+        pool_name: name of the pool
+        image_name: name of the image
+        cache_type: type of cache i.e dm-cache and dm-write cache
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+
+    """
+    config = kw.get("config")
+    device_names = []
+    lv_sizes = {"data_disk": "4G", "cache_disk": "2G", "meta_disk": "2G"}
+
+    if cache_type == "cache":
+        vg_name = "test_cache_vg"
+        test_folder_path = "/tmp/test_cache"
+        VG_disk = "/dev/test_cache_vg/data_disk"
+    else:
+        vg_name = "test_write_cache_vg"
+        test_folder_path = "/tmp/test_write_cache"
+        VG_disk = "/dev/test_write_cache_vg/data_disk"
+
+    device_names.append(rbd_obj.image_map(pool_name, image_name)[:-1])
+    pvcreate(client_node, device_names[0])
+    vgcreate(client_node, vg_name, device_names[0])
+
+    for name, size in lv_sizes.items():
+        lvm_create(client_node, name, vg_name, size)
+
+    if cache_type == "cache":
+        # to configure dm-cache pool in LVM by associating a metadata disk with a cache disk
+        command = f"echo 'y' | lvconvert --type cache-pool --poolmetadata {vg_name}/meta_disk {vg_name}/cache_disk"
+        client_node.exec_command(cmd=command, sudo=True)
+
+        lvconvert(client_node, cache_type, vg_name, "cache_disk", "data_disk")
+    else:
+        # Configure dm-write cache
+        command = f"lvchange --activate n {vg_name}/cache_disk"
+        client_node.exec_command(cmd=command, sudo=True)
+
+        lvconvert(client_node, cache_type, vg_name, "cache_disk", "meta_disk")
+
+    # creating XFS filesystem on VG_disk
+    run_mkfs(client_node=client_node, device_name=VG_disk, type="xfs")
+
+    command = f"mkdir -p {test_folder_path}"
+    client_node.exec_command(cmd=command, sudo=True)
+
+    # mount the cached disk and run fio
+    command = f"mount {VG_disk} {test_folder_path}"
+    client_node.exec_command(cmd=command, sudo=True)
+
+    run_fio(
+        client_node=client_node,
+        device_name=VG_disk,
+        runtime=config.get("runtime", 30),
+    )
+
+    # check ceph health
+    out, error = client_node.exec_command(cmd="ceph -s", sudo=True)
+
+    if "HEALTH_ERR" in out:
+        log.debug(out)
+        log.error("Cluster went to error health state")
+        return 1
+
+    log.info("Starting to cleanup cache drive")
+    client_node.exec_command(cmd=f"wipefs -af {device_names[0]}", sudo=True)
```

Comment:
This cleanup will not execute when there is a HEALTH_ERR right? Which might not be desirable can we do cleanup when there is health error.

Reply:
Maybe add wipefs in finally.

Reply:
Done @manasagowri Please review it.